### DISCUSSION TITLE
understanding composition model

### DISCUSSION THREAD

**Discussion by naveenmohandas** [CederGroupHub/chgnet#180] [General]
Hi, 
I am looking at the fine-tuning of the model. In the example given for fine-tuning the model there is a section describing **Atomref**. I understand why we need it, but I do not understand how it is used in the model? Would it be possible to elaborate on how the composition model works and what it does during training? 
I am also available for a short meeting if it would be difficult to explain in words.

**Comment by bowen-bd** [comment]
MLIPs like CHGNet are designed to learn the interaction between atoms.
And there exist non-zero energies of non-interacting (isolated) atoms.
We leave this part for the `Atomref` to learn, so the GNN can dedicate to learn the remaining part.
Please also refer to the centering concept in data science

**Reply by naveenmohandas** [reply]
Thank you for the quick reply. I have two more questions regarding it: 
1. I understand how it would work for structures with just one element, but I find it difficult to understand how it would correct for structures with more than one element. Could you explain how the regression works?
2. Also, does it that mean all energies are shifted by the `Atomref` energies before it is fed to the model for training?

**Comment by naveenmohandas** [comment]
Hi, 
While fine-tuning I get some weird behaviours, so could you also let me know if I am passing the composition model the right way for fine-tuning? or is there some other way I need to pass the composition model?

```
new_atom_ref = AtomRef(is_intensive=True)
new_atom_ref.initialize_from_MPtrj()
new_atom_ref.fit(structures, energies_per_atom)

model = CHGNet.load()
model.composition_model = new_atom_ref
```
